tests_e2e/test_browser.py

```diff
@@ -0,0 +1,163 @@
+from datetime import date
+import pytest
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+import os
+from threading import Thread
+from todo_app import app
+from todo_app.board import Board
+
+CHROME_DRIVER_PATH = '/path/to/your/chromedriver'
+
+
+@pytest.fixture(scope='module')
+def driver():
+    opts = webdriver.ChromeOptions()
+    opts.add_argument('--headless')
+    with webdriver.Chrome(CHROME_DRIVER_PATH, options=opts) as driver:
+        yield driver
+
+
+@pytest.fixture(scope='module')
+def app_with_temp_board():
+    os.environ['TRELLO_BOARD_NAME'] = 'DummyBoard'
+    board_id = Board.create_board(os.environ.get('TRELLO_API_KEY'), os.environ.get(
+        'TRELLO_TOKEN'), os.environ.get('TRELLO_BOARD_NAME'))
+
+    application = app.create_app()
+
+    # start the app in its own thread.
+    thread = Thread(target=lambda: application.run(use_reloader=False))
+    thread.daemon = True
+    thread.start()
+    yield application
+
+    # Tear Down
+    thread.join(1)
+    Board.delete_board(os.environ.get('TRELLO_API_KEY'),
+                       os.environ.get('TRELLO_TOKEN'), board_id)
+
+
+def test_tasks_are_empty_on_startup(driver, app_with_temp_board):
+    driver.get('http://localhost:5000/')
+    assert 'To-Do App' in driver.title
+
+
+def test_add_tasks(driver, app_with_temp_board):
+    driver.get('http://localhost:5000/')
+    wait = WebDriverWait(driver, 10)
+
+    title_field = driver.find_element_by_id("title")
+    title_field.send_keys('Do grocery shopping')
+    title_field.submit()
+
+    new_todo_row = wait.until(
+        EC.visibility_of_element_located((By.ID, 'not-started-1')))
+
+    assert 'Do grocery shopping' in new_todo_row.text
+
+    title_field = driver.find_element_by_id("title")
+    description_field = driver.find_element_by_id("description")
+    due_date_field = driver.find_element_by_id("due-date")
+
+    future_year = date.today().year + 1
+    future_date = f'19/03/{future_year}'
+
+    title_field.clear()
+    title_field.send_keys('Take a nap')
+    description_field.clear()
+    description_field.send_keys('Sleep for at least 30 minutes')
+    due_date_field.clear()
```

Comment:
This test errored for me at this stage with a complaint this field wasn't user editable - I haven't looked in that far but removing the clear step resolved it and feels reasonable here

It's worth noting that this caused other of your tests to fail - having these tests so interdependent (e.g. relying on each otherss setup) could cause you pain in future